test/goalc/test_arithmetic.cpp

```diff
@@ -0,0 +1,232 @@
+  }
+
+  return tests;
+}
+
+// In the interest of speed, we want to share the same thread/compiler across
+// all the tests in this suite, so we have to over-ride this.
+class ArithmeticTests : public testing::TestWithParam<IntegerParam> {
+ public:
+  // Per-test-suite set-up.
+  // Called before the first test in this test suite.
+  static void SetUpTestSuite() {
+    runtime_thread = std::thread((GoalTest::runtime_no_kernel));
+    runner.c = &compiler;
+  }
+
+  // Per-test-suite tear-down.
+  // Called after the last test in this test suite.
+  static void TearDownTestSuite() {
+    compiler.shutdown_target();
+    runtime_thread.join();
+  }
+
+  // You can define per-test set-up logic as usual.
+  void SetUp() {
+    GoalTest::createDirIfAbsent(GoalTest::getTemplateDir(testCategory));
+    GoalTest::createDirIfAbsent(GoalTest::getGeneratedDir(testCategory));
+  }
+
+  // You can define per-test tear-down logic as usual.
+  void TearDown() {}
+
+  // Common Resources Across all Tests in the Suite
+  static std::thread runtime_thread;
+  static Compiler compiler;
+  static GoalTest::CompilerTestRunner runner;
+
+	// Just to promote better test organization, supports nesting the test files 1 directory deep
+	std::string testCategory = "arithmetic";
+  inja::Environment env{GoalTest::getTemplateDir(testCategory), GoalTest::getGeneratedDir(testCategory)};
+};
+
+// You must initialize the static variables outside of the declaration, or you'll run into unresolved external errors
+std::thread ArithmeticTests::runtime_thread;
+Compiler ArithmeticTests::compiler;
+GoalTest::CompilerTestRunner ArithmeticTests::runner;
+
+
+// Finally, we define our generic test, given our custom class that represents our test inputs
+// we can generate the lisp file, and pass along the path to the test runner
+// If the test fails, the test runner will save the template file, with the expected/actual results into the `failed/` directory
+TEST_P(ArithmeticTests, EvalIntegers) {
+  IntegerParam param = GetParam();
+  nlohmann::json data;
+  data["integer"] = param.toLisp();
+
+  std::string testFile = "eval-integer-" + std::to_string(param.index) + ".generated.gc";
+  env.write("eval-integer.template.gc", data, testFile);
+  runner.run_test(testCategory, testFile, {param.eval()});
+}
+
+// ValuesIn, is not the only way to use a parameterized test, but the most applicable for this example
+// You can actually get googletest to compute the permutations for you, which may be useful.  Consult their docs.
+// - https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests
+INSTANTIATE_TEST_SUITE_P(EvalIntegers,
+                         ArithmeticTests,
+                         testing::ValuesIn(genIntegerTests(4,
+                                                           true,
+                                                           true,
+                                                           {IntegerParam(-2147483648),
```

Comment:
At some point we should add in more edge case integers.  See 'IR_LoadConstant64::do_codegen` which picks one of 5 different instructions for doing the load depending on the size/sign of the value.